### ISSUE TITLE
How to access the response headers?

### ISSUE THREAD

**Issue by broodfusion** [aribouius/jsonapi-react#15]
First of all, awesome library!

My use case is that I am refreshing jwt token on API requests, the new jwt token is set on the response header.
Is there a way to access the response header? 

Thank you!

**Comment by aribouius** [comment]
Hey @broodfusion my apologies, I saw your comment a while back and could have sworn I responded to it, but guess not.  Right now there's unfortunately not a great way to access a response header, as the library abstracts the request cycle quite a bit.

Passing the response headers through `useQuery()` or the `client.fetch()` doesn't appear viable at the moment, as it would be a large breaking change.  I'd love to support this somehow though, and finally have some time to dig into it now.

The `client` instance does implement a basic event subscription system ([see here](https://github.com/aribouius/jsonapi-react/blob/master/src/client.js#L201)), so I was thinking a decent solution might be to dispatch a new event whenever a request is received, and pass through actual response object.

Thoughts?
